vpr/src/route/heap_type.h

```diff
@@ -6,6 +6,24 @@
 #include "vtr_memory.h"
 #include "vtr_array_view.h"
 #include "rr_graph_fwd.h"
+#include <set>
+
+/* Extra path data needed by RCV, seperated from t_heap struct for performance reasons
+ * Can be accessed by a pointer, won't be initialized unless needed
+ *
+ * path_rr: The entire partial path up until the route tree
+ * 
+ * edge: A list of edges from each node in the partial path to reach the next node
+ * 
+ * backward_delay: The delay of the partial path plus the path from route tree to source
+ * 
+ * backward_cong: The congestion estimate of the partial path plus the path from route tree to source */
+struct t_heap_path {
+    std::vector<int> path_rr;
+    std::vector<int> edge;
```

Comment:
If you can change this vector to ```std::vector<RRNodeId>```, that'd be great for type checking.
If the edge also has a similar strongId type(I think ```RREdgeId```?), then changing that from int to the special Id type would be beneficial as well.

Reply:
Alright sure that's not a bad idea, but RRNodeId isn't just a typedef so I might quickly check what the size differences are between int and RRNodeId, as it could marginally effect performance.

Reply:
I think the strong id class is sufficiently optimized so that it's widely used. Also if you have used strong id here, you'd be able to find the bug I pointed out by a simple compilation. Anyway, I'd recommend using strong ids since it is more important than speed for various reasons.

Reply:
### It's really an int with extra type checking at compile time.  I'm pretty sure there is no run time overhead (Kevin would have checked when he built it; if you convert you can always run a design to double check but I'm pretty confident you won't notice any speed degradation).